core/store/models/eth.go

```diff
@@ -200,6 +204,9 @@ func (f *FunctionSelector) UnmarshalJSON(input []byte) error {
 	}
 
 	if utils.HasHexPrefix(s) {
+		if !hexRegexp.Match([]byte(s)) {
```

Comment:
There is some redundancy here, `HasHexPrefix` checks for `0x` prefix, but your Regex treats that as optional.

Reply:
Thanks, fixed in 75beb41750c96dc6a9eda67028b5f4cbbd0c95a5